backend/src/main/java/com/step2020/server/managers/EventCreationManager.java

```diff
@@ -70,6 +72,237 @@ public EventCreationManager(String sessionId) {
+	  Map<String, String> response = Utility.createResponse("failed", errorMessage);
+	  Utility.sendResponseAndRemoveRequest(sessionId, requestId, response);
+	}
+      }
+      public void onCancelled(DatabaseError error) {
+	// Send failed information
+	String errorMessage = error.getMessage();
+	Map<String, String> response = Utility.createResponse("failed", errorMessage);
+	Utility.sendResponseAndRemoveRequest(sessionId, requestId, response);
+      }	
+    });
+  }
+  
+  // Updates the event with the map info with just the info needed to be updated
+  private void updateEventWithMapInfo(String requestId, String eventId, Map<String, String> updateInfo) {
+    // Change to updated info into a map of String keys and Object values so Firebase can handle input
+    Map<String, Object> updateInfoObject = Collections.<String, Object>unmodifiableMap(updateInfo);
+
+    // Query the event id and update the children with the update information
+    this.eventsRef.child(EVNTS).child(eventId).updateChildren(updateInfoObject, new DatabaseReference.CompletionListener() {
+      @Override
+      public void onComplete(DatabaseError databaseError, DatabaseReference databaseReference) {
+        // If there aren't any errors, send in a success response, if not, send in failed error
+	if (databaseError == null) {
+	  Map<String, String> response = Utility.createResponse("success", "");
+	  Utility.sendResponseAndRemoveRequest(sessionId, requestId, response);
+	} else {
+	  String errorMessage = databaseError.getMessage();
+	  Map<String, String> response = Utility.createResponse("failed", errorMessage);
+	  Utility.sendResponseAndRemoveRequest(sessionId, requestId, response);
+	}
+      }
+    });
+  } 
+
+  // Deletes the event with the requested event id in the requestInfo map and make sure the user is the owner
+  public void deleteEvent(String requestId, Map<String, String> requestInfo) {
+    // Check to make sure the requestinfo has the required keys
+    if (!requestInfo.containsKey("uid") && !requestInfo.containsKey("eventId")) {
+      String errorMessage = "Required fields are empty: uid and eventId.";
+      Map<String, String> response = Utility.createResponse("failed", errorMessage);
+      Utility.sendResponseAndRemoveRequest(sessionId, requestId, response);
+    }
+    
+    // Get the uid and event id
+    String uid = requestInfo.get("uid");
+    String eventId = requestInfo.get("eventId"); 
+
+    System.out.println(eventId);
+    // Query the event and owner id
+    this.eventsRef.child(EVNTS).child(eventId).addListenerForSingleValueEvent(new ValueEventListener() {
+      public void onDataChange(DataSnapshot snapshot) {
+	String ownerId = snapshot.child("ownerId").getValue(String.class);
+	// Check to make sure the ownerId and the uid are the same
+	if (ownerId.equals(uid)) {
+	  // remove event
+	  removeEvent(requestId, eventId);
+	} else {
+	  String errorMessage = "You are not the owner of this event.";
```

Comment:
Also could be a helper function - `handleUnownedEventActionForRequestID` or something